bitnami/opensearch/templates/coordinating/metrics-svc.yaml

```diff
@@ -0,0 +1,51 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if and ( include "opensearch.coordinating.enabled" . ) .Values.coordinating.metrics.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ printf "%s-metrics" (include "opensearch.coordinating.fullname" .) | trunc 63 | trimSuffix "-" }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/part-of: opensearch
+    app.kubernetes.io/component: coordinating-only
+  {{- $defaultAnnotations := dict "prometheus.io/scrape" "true" "prometheus.io/port" (.Values.coordinating.metrics.service.port | quote) "prometheus.io/path" "/_prometheus/metrics" }}
+  {{- $annotations := include "common.tplvalues.merge" (dict "values" (list $defaultAnnotations .Values.coordinating.metrics.service.annotations .Values.commonAnnotations) "context" .) }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $annotations "context" $ ) | nindent 4 }}
+spec:
+  type: {{ .Values.coordinating.metrics.service.type }}
+  {{- if and .Values.coordinating.metrics.service.clusterIP (eq .Values.coordinating.metrics.service.type "ClusterIP") }}
+  clusterIP: {{ .Values.coordinating.metrics.service.clusterIP }}
+  {{- end }}
+  {{- if (or (eq .Values.coordinating.metrics.service.type "LoadBalancer") (eq .Values.coordinating.metrics.service.type "NodePort")) }}
+  externalTrafficPolicy: {{ .Values.coordinating.metrics.service.externalTrafficPolicy | quote }}
+  {{- end }}
+  {{- if eq .Values.coordinating.metrics.service.type "LoadBalancer" }}
+  loadBalancerSourceRanges: {{ .Values.coordinating.metrics.service.loadBalancerSourceRanges }}
+  {{- end }}
+  {{- if (and (eq .Values.coordinating.metrics.service.type "LoadBalancer") (not (empty .Values.coordinating.metrics.service.loadBalancerIP))) }}
+  loadBalancerIP: {{ .Values.coordinating.metrics.service.loadBalancerIP }}
+  {{- end }}
+  {{- if .Values.coordinating.metrics.service.sessionAffinity }}
+  sessionAffinity: {{ .Values.coordinating.metrics.service.sessionAffinity }}
+  {{- end }}
+  {{- if .Values.coordinating.metrics.service.sessionAffinityConfig }}
+  sessionAffinityConfig: {{- include "common.tplvalues.render" (dict "value" .Values.coordinating.metrics.service.sessionAffinityConfig "context" $) | nindent 4 }}
+  {{- end }}
```

Comment:
I find this overcomplicated for metrics.. I'd assume metrics services are always `ClusterIP` ones